Component tokens and lots of small bug fixes

twjeffery opened the pull request.

twjeffery commented on libs/web-components/src/components/icon/Icon.svelte:
```diff
@@ -1,7 +1,7 @@
 
 <script lang="ts" context="module">
-  export type IconSize = "small" | "medium" | "large" | "xlarge";
+  export type IconSize = "1" | "2" | "3" | "4" | "5" | "6" | "2xsmall" | "xsmall" | "small" | "medium" | "large" | "xlarge";
```
I added a new icon size scale (size 1-6) because the other (tshirt sizing) are misaligned with the token values. (i.e. goa-icon-small was using icon-size-medium) 

This allow us to easily add the new icon sizes and fix this misalignment going forward without breaking changes.

twjeffery replied:
I added the additional types to the common.ts file, and I'm now referencing it in the icon.svelte file instead, same with the `IconTheme` and `IconVariant`

Added test coverage for each size to icon.spec.ts

twjeffery commented on libs/web-components/src/components/badge/Badge.svelte:
```diff
@@ -96,63 +96,94 @@
+    padding: var(--goa-badge-padding);  /* is calc(3 / 16 * 1rem) better for scaling? */
+    gap: var(--goa-badge-gap);
     font-weight: var(--goa-font-weight-regular);
+    /*box-shadow: var(--goa-badge-border);   / inner shadow - new style */
```
This is the new border style for badges as part of the upcoming style updates. Commented out for now.

twjeffery commented on libs/web-components/src/components/button/Button.svelte:
```diff
@@ -152,134 +220,122 @@
-      --font-valign-fix,
-      0
-    ); /* acumin font requires this to allow for vertical alignment  */
+    padding-bottom: 0.2rem; /* acumin font requires this to allow for vertical alignment  */
```
font-v-align fix is not completely centering text, replaced with manual value

twjeffery commented on libs/web-components/src/components/button/Button.svelte:
```diff
@@ -84,64 +84,132 @@
       <slot />
     </span>
-    <goa-icon id="trailing-icon" type="arrow-forward" inverted="true" />
+    <goa-icon id="trailing-icon" size="4" type="arrow-forward" inverted="true" />
```
button changed to be different sizes in start, regular, and compact buttons. 

I was not sure how to add a difference in the compact button. Added token value and class below.

twjeffery replied:
added test for size token in spec.ts

twjeffery commented on libs/web-components/src/components/footer/Footer.svelte:
```diff
@@ -48,13 +50,15 @@
-            src="data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink' width='149.351' height='42' viewBox='0 0 149.351 42'%3E%3Cdefs%3E%3Cstyle%3E.a%7Bfill:none;%7D.b%7Bclip-path:url(%23a);%7D.c%7Bfill:%2300aad2;%7D.d%7Bfill:%235f6a72;%7D%3C/style%3E%3CclipPath id='a'%3E%3Crect class='a' width='149.351' height='42'/%3E%3C/clipPath%3E%3C/defs%3E%3 ...
-          />
+      <!-- Logo and optional link -->
+      {#if url}
```
Added component property to set custom link for logo. If no url set, then no link on logo. (just like we have for app header logo)

BumbleB2na replied:
In my opinion the logo in footer should always link to https://www.alberta.ca without letting user change it. Take this comment with a grain of salt and will share more with you offline about my opinion on this.

twjeffery replied:
I see what you mean. The point was for teams to be able to optionally change the url or disable it as needed. 

I've adjusted this to default to [Alberta.ca](http://alberta.ca), and then by adding a url="http...." can change to a custom link, and by adding url="" can disable the link.

BumbleB2na replied:
Since `url=""` disables it then I'm not sure if defaulting to a url makes sense anymore. Might be better to leave it empty and disabled by default.

twjeffery replied:
I agree with you, but that would mean a change for teams currently using it (it would remove a link that is there now. With what I have implemented, nothing would change currently and then they have the option to change the url from the default or remove it.

twjeffery commented on libs/web-components/src/components/footer/Footer.svelte:
```diff
@@ -48,13 +50,15 @@
+      <!-- Logo and optional link -->
+      {#if url}
+        <a href={url}>
+          <img src={goalogo} alt="Government of Alberta Logo" class="logo"/>
```
I updated the logo svg (updated colours from CPE).

Would we rather have the svg embedded for performance/loading reasons instead of linked to a file? I can change this.

twjeffery commented on libs/web-components/src/components/footer/Footer.svelte:
```diff
@@ -83,31 +102,41 @@
 
   @container self (--mobile) {
     .content {
```
added mobile variants for all footers with breakpoint (different type size and spacing as per design)

twjeffery commented on libs/web-components/src/components/footer/Footer.svelte:
```diff
@@ -83,31 +102,41 @@
   @container self (--mobile) {
     .content {
-      padding: 2rem 1rem;
+      padding: var(--goa-footer-padding-small-screen);
```
added mobile variants for all footers with breakpoint (different type size and spacing as per design)

twjeffery commented on libs/web-components/src/components/tabs/Tabs.svelte:
```diff
@@ -221,70 +221,110 @@
+    border-bottom: var(--goa-tab-border-hover);
   }
 
+  :global([role="tabpanel"]:focus-visible) {
```
added focus styling for tab content

twjeffery replied:
Before

<img width="707" alt="image" src="https://github.com/user-attachments/assets/23a93a85-86ad-495c-b8e2-0643f093b0aa">


After

<img width="1033" alt="image" src="https://github.com/user-attachments/assets/dbf3dd83-0f6a-4467-ae57-7b46f5dc97de">

twjeffery commented on libs/web-components/src/components/tab/Tab.svelte:
```diff
@@ -79,7 +79,4 @@
 </section>
 
 <style>
-  [role="tabpanel"] {
```
moved to tabs component

twjeffery commented on libs/web-components/src/components/tabs/Tabs.svelte:
```diff
@@ -221,70 +221,110 @@
   @media (--mobile) {
+
+    .tabs {
+      border-left: var(--goa-tabs-bottom-border);
```
added left and bottom border as per design. 

<img width="551" alt="image" src="https://github.com/user-attachments/assets/410bfd83-bb7b-4fce-b389-97ed1e09ab86">

twjeffery commented on libs/web-components/src/components/tabs/Tabs.svelte:
```diff
@@ -221,70 +221,116 @@
-      border-left: 4px solid transparent;
+      padding: var(--goa-tab-padding-mobile);
+      border-left: var(--goa-tab-border-not-selected);
+      text-overflow: ellipsis;
```
added ellipses when text overflow

added min width for tab, and horizontally centred text for min width.

twjeffery commented on libs/web-components/src/components/modal/Modal.svelte:
```diff
@@ -233,10 +233,12 @@
                 <!-- svelte-ignore a11y-click-events-have-key-events -->
                 <!-- svelte-ignore a11y-no-static-element-interactions -->
                 <goa-icon-button
+                  size="medium"
```
fixed close icon button size and thicker icon.

twjeffery commented on libs/web-components/src/components/modal/Modal.svelte:
```diff
@@ -344,26 +366,61 @@
+    flex-direction: column;
+    }
+
+    .callout-bar {
```
added mobile callout variant -- with top bar layout (as per design)

<img width="551" alt="image" src="https://github.com/user-attachments/assets/1f53e7c1-ba47-4ec4-9d5d-ab9a628d8a94">

twjeffery commented on libs/web-components/src/components/checkbox/Checkbox.svelte:
```diff
@@ -195,17 +216,26 @@
   }
 
+  /* Add hover styles when the user hovers over the label */
+  label:hover .container {
```
added hover when user hovers over label (not just checkbox square)

twjeffery commented on libs/web-components/src/components/radio-item/RadioItem.svelte:
```diff
@@ -47,7 +47,7 @@
   export let mt: Spacing = null;
   export let mr: Spacing = null;
-  export let mb: Spacing = "m";
+  export let mb: Spacing = null;
```
moved the margin-bottom on radio item to gap on radio group, to fix extra space between helper text and radio group (bug)

twjeffery commented on libs/web-components/src/components/radio-item/RadioItem.svelte:
```diff
@@ -181,116 +217,116 @@
   .goa-radio-label {
-    padding: 0 var(--goa-space-xl) 0 var(--goa-space-xs);
-    font-weight: var(--goa-font-weight-regular);
+    padding: 0 var(--goa-space-xs);
```
moved padding right on the item to GAP on the group when horizontal. This fixes a weird bug where the space right beside the next radio input is clickable

twjeffery commented on libs/web-components/src/components/radio-item/RadioItem.svelte:
```diff
@@ -181,116 +217,116 @@
-  .goa-radio--disabled .goa-radio-label {
-    opacity: 0.4;
+  .goa-radio--disabled .goa-radio-label,
+  .goa-radio--disabled ~ .goa-radio-description {
```
fixed item descriptions not ”disabling” when input is disabled

twjeffery commented on libs/web-components/src/components/popover/Popover.svelte:
```diff
@@ -304,17 +313,17 @@
     margin-bottom: var(--offset-bottom, 3px);
     margin-left: var(--offset-left, 0);
     margin-right: var(--offset-right, 0);
+    /* overflow: hidden; */
```
This would clip the background borders radius in the app-header menu when hovering menu items.

twjeffery replied:
<img width="551" alt="image" src="https://github.com/user-attachments/assets/273bdd67-8750-44c9-ac65-1dc357618c80">
<img width="551" alt="image" src="https://github.com/user-attachments/assets/3b5c52c4-2638-426f-89b9-00c67532ea2f">

twjeffery commented on libs/web-components/src/components/tooltip/Tooltip.svelte:
```diff
@@ -234,26 +246,34 @@
     align-items: center;
   }
 
+  .tooltip:focus-visible {
```
moved the tooltip gap to the tooltip, instead of the focus, because this happens:

Before
![image](https://github.com/user-attachments/assets/39d7826e-5bea-4207-991c-d1cf2ce444e4)

After
<img width="699" alt="image" src="https://github.com/user-attachments/assets/6f78c919-eb75-46f6-b700-8b62fa4566ce">

twjeffery commented on libs/web-components/src/components/app-header/AppHeader.svelte:
```diff
@@ -19,11 +19,11 @@
 
   const _mobileLogo =
-    "data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink' width='32' height='32' viewBox='0 0 16 16'%3E%3Cdefs%3E%3Cstyle%3E.a%7Bfill:none;%7D.b%7Bfill:%2300aad2;%7D.c%7Bclip-path:url(%23a);%7D.d%7Bfill:%23fff;%7D%3C/style%3E%3CclipPath id='a'%3E%3Crect class='a' width='14' height='14' transform='translate(-0.345 -0.21)'/%3E%3C/clipPath%3E%3C/def ...
+    "data:image/svg+xml,%3Csvg width='24' height='24' viewBox='0 0 24 24' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cg clip-path='url(%23clip0_169_613)'%3E%3Cpath d='M12 0C14.3734 0 16.6935 0.703788 18.6668 2.02236C20.6402 3.34094 22.1783 5.21509 23.0866 7.4078C23.9948 9.60051 24.2324 12.0133 23.7694 14.3411C23.3064 16.6689 22.1635 18.8071 20.4853 20.4853C18.8071 22.1635 16.6689 23.3064  ...
```
updated logo

twjeffery commented on libs/react-components/src/lib/footer/footer.tsx:
```diff
@@ -3,6 +3,7 @@ import { ReactNode } from "react";
 interface WCProps {
   maxcontentwidth?: string;
   testid?: string;
+  url?: string;
```
added property in footer component for url on logo

twjeffery replied:
Added `@Input() url?: string;` to footer.ts

twjeffery commented on libs/web-components/src/components/notification/Notification.svelte:
```diff
@@ -99,54 +130,64 @@
   }
 
+  /* Screen sizes */
+
```
Fixed alignment of icon and icon thickness

Before

<img width="690" alt="image" src="https://github.com/user-attachments/assets/eb347291-5e2f-48d5-a453-bbfaf90a39f9">

After

<img width="962" alt="image" src="https://github.com/user-attachments/assets/689ea9f3-db4c-4162-9b6e-27bfd928b5a2">

twjeffery commented on libs/web-components/src/components/footer/Footer.svelte:
```diff
@@ -164,6 +197,16 @@
+    color: var(--goa-footer-color-links-hover);
+  }
+
+  a:focus-visible {
```
Added hover and focus styling for links in the footer

twjeffery commented on libs/web-components/src/components/microsite-header/MicrositeHeader.svelte:
```diff
@@ -115,30 +131,51 @@
   }
 
+  .feedback-link {
+    display: inline-flex; /* Keeps the text and icon together when text wraps */
```
fixed wrapping on external icon and text (stays together)

twjeffery commented on libs/web-components/src/components/side-menu/SideMenu.svelte:
```diff
@@ -110,31 +110,81 @@
+    height: 100%;
+    flex-direction: column;
+    gap: var(--goa-side-menu-items-gap);
+    border-right: var(--goa-side-menu-border-right); /* should this happen on the parent div on the page? */
```
added right border (in design)

twjeffery commented on libs/web-components/src/components/side-menu/SideMenu.svelte:
```diff
@@ -110,31 +110,81 @@
-    display: block;
+    width: var(--goa-side-menu-width);
+    display: flex;
+    height: 100%;
```
Added 100% height

twjeffery commented on libs/web-components/src/components/skeleton/Skeleton.svelte:
```diff
@@ -310,10 +314,11 @@
     display: flex;
     flex-direction: row;
     align-items: center;
+    gap: 1rem;
```
Fixed bug on "profile" type.

Before

<img width="489" alt="image" src="https://github.com/user-attachments/assets/c18b1430-4c4d-47e7-9ca2-fc4dad6a1cd4">


After

<img width="624" alt="image" src="https://github.com/user-attachments/assets/218042fb-0371-4d1f-bb27-143fc0fc17b5">

twjeffery commented on libs/web-components/src/components/modal/Modal.svelte:
```diff
@@ -275,6 +277,25 @@
 <!-- ======================================================================= -->
 <style>
   :host {
+    /* TODO: Component tokens, to move to design tokens file ------------------------------------------------------- */
```
Fixed various spacing bugs

Before

<img width="781" alt="image" src="https://github.com/user-attachments/assets/ded84723-3c6f-48ee-ab5c-f37e4504bf18">
<img width="781" alt="image" src="https://github.com/user-attachments/assets/bc593eb7-3577-4889-989c-9dd7ab6b98f0">


After

<img width="781" alt="image" src="https://github.com/user-attachments/assets/7972ea48-c7ce-40d4-8156-7b15f8da3cab">
<img width="781" alt="image" src="https://github.com/user-attachments/assets/799b2839-924d-4e96-91f0-d597e58df10a">

twjeffery replied:
Fixed a lot of issues with the scroll bar, buttons on mobile, and spacing overall when scroll and when on mobile.

Before

<img width="702" alt="image" src="https://github.com/user-attachments/assets/9a0e7e8a-d1cd-4933-8e41-d3b6110edd52">

After
<img width="534" alt="image" src="https://github.com/user-attachments/assets/5a2b2bb1-d863-48dd-9e4e-ac7f4e6d05c8">

twjeffery commented on libs/web-components/src/components/checkbox/Checkbox.svelte:
```diff
@@ -239,23 +269,78 @@
     background-color: var(--goa-color-greyscale-white);
     box-shadow: none;
   }
+  .error .container.selected,
```
Fixed error states on hover and focus

twjeffery commented on libs/web-components/src/components/checkbox/Checkbox.svelte:
```diff
@@ -274,7 +359,16 @@
+  .disabled.error .container {
+    border: var(--goa-border-width-m) solid #F58185;
   }
+  label:hover.disabled.error .container {
```
Fixed: description not changing colour on disabled state

Before

<img width="297" alt="image" src="https://github.com/user-attachments/assets/fa07a4d7-9a18-4789-9aff-8bda346fb39e">

After

<img width="408" alt="image" src="https://github.com/user-attachments/assets/e3e5a381-c829-491c-bd74-9b89c383f1e1">

twjeffery commented on libs/web-components/src/components/radio-group/RadioGroup.svelte:
```diff
@@ -175,16 +175,25 @@
   .goa-radio-group--horizontal {
     display: flex;
     flex-direction: row;
+    gap: var(--goa-radio-group-gap-horizontal);
```
- moved padding right on the item to GAP on the group when horizontal
    - this also fixes a weird bug where the space right beside the next radio input is clickable

twjeffery commented on libs/web-components/src/components/scrollable/Scrollable.svelte:
```diff
@@ -43,31 +43,40 @@
 
 <style>
+
+
```
Styled the scroll bar

twjeffery commented on libs/web-components/src/assets/css/components.css:
```diff
@@ -57,7 +57,10 @@ goa-table[variant="relaxed"] td {
 goa-table thead th {
```
Can we move the styles for the table to the table svelte file?

twjeffery replied:
ok, I will leave them as is for now

twjeffery commented on libs/web-components/src/components/table/TableSortHeader.svelte:
```diff
@@ -55,8 +55,8 @@
-    height: 3.75rem;
-    padding: 0 1rem;
+    height: 3rem;
+    padding: 12px 1rem;
```
fixed bug with layout and hover

Before
<img width="583" alt="image" src="https://github.com/user-attachments/assets/2e80184d-07e7-42b7-94f9-9bdc81f97598">


After
<img width="583" alt="image" src="https://github.com/user-attachments/assets/8fe00769-60bc-4c69-aa00-e72d83a82ed3">

twjeffery commented on libs/web-components/src/components/microsite-header/MicrositeHeader.svelte:
```diff
@@ -71,6 +71,20 @@
           href="https://www.alberta.ca/index.aspx"
           target={`_${headerurltarget}`}>Alberta Government</a
         >
+        {#if feedbackurl}
```
<img width="954" alt="image" src="https://github.com/user-attachments/assets/01119f30-b676-42ee-8796-dbc51998fc58">

twjeffery replied:
added in tests for this in microsite header spec.ts file.

twjeffery commented on libs/web-components/src/components/file-upload-input/FileUploadInput.svelte:
```diff
@@ -254,56 +254,90 @@
-
-  .state-default {
-    background: var(--goa-color-info-background);
+  .dragdrop:focus-within {
```
updated focus state

![image](https://github.com/user-attachments/assets/a60e9620-175a-47fd-a74b-d760a656e8ee)
![image](https://github.com/user-attachments/assets/c89d71e5-5527-4350-9f44-1a82d454008d)

twjeffery commented:
Here are my notes on all of the small bug fixes: https://goa-dio.atlassian.net/wiki/spaces/DS/pages/3552313784/A+lot+of+little+bug+fixes

ArakTaiRoth commented:
- [ ] Default URL removed from AppFooter, if no URL is provided now the logo doesn't link to anything
- [ ] Adding a link to the AppFooter, increases the height of the image by 7px. This happens on our current AppFooter as well, but there you can't remove the URL, so it's not apparent
- [ ] Badge when using an icon gets moved down slightly. I can't seem to find where the extra space is coming from, nothing changes in total height or anything.
![badge](https://github.com/user-attachments/assets/6984ca76-d549-4eec-bed7-d78ea9267394)
- [ ] New `inverse` variant on Button isn't usable in React. Also `inverse` wasn't added to the `html-data.json` either
- [ ] If using `filterable` for Dropdown, it overwrites whatever the user might have for `leadingicon`

twjeffery commented:
@ArakTaiRoth 

> * [ ]  Default URL removed from AppFooter, if no URL is provided now the logo doesn't link to anything
- that's as intended, they can choose to set a URL, and if none set, no URL. Same as how our header url currently works for the logo in the header.
> * [ ]  Adding a link to the AppFooter, increases the height of the image by 7px. This happens on our current AppFooter as well, but there you can't remove the URL, so it's not apparent
- Just pushed a fix for this in another commit by adding display:flex to `a` in footer, now same size with and without link.
> * [ ]  Badge when using an icon gets moved down slightly. I can't seem to find where the extra space is coming from, nothing changes in total height or anything. Seems like this is an issue with current production, so not sure if we want to fix here or not
>   ![badge](https://private-user-images.githubusercontent.com/1081991/401284771-6984ca76-d549-4eec-bed7-d78ea9267394.PNG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYzNjYzMTAsIm5iZiI6MTczNjM2NjAxMCwicGF0aCI6Ii8xMDgxOTkxLzQwMTI4NDc3MS02OTg0Y2E3Ni1kNTQ5LTRlZWMtYmVkNy1kNzhlYTkyNjczOTQuUE5HP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDEwOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAxMDhUMTk1MzMwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OGZkNTQ2ZTc2MTg2Nzk4NDkyZmJhY2Q3ZTJmZGQ4YjY2ZTljNGIzMjA3Nzc2OTJlYjQ4ZGRmZWMxMDAyNjViMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.cD_5SpIdXEq6bJMCJ75a-9Crpmh5LjkmkXjezoW_6po)
- I fixed the badge height issue. Included in new commit.
> * [ ]  New `inverse` variant on Button isn't usable in React. Also `inverse` wasn't added to the `html-data.json` either
- this should probably be removed and added in a different story? I created a first version of an inverse variant for the button styling. Removed "inverse" type from code in latest commit.
> * [ ]  If using `filterable` for Dropdown, it overwrites whatever the user might have for `leadingicon`
- I thought that adding the leading icon automatically for filterable was a good idea, but maybe not (at least for now), especially as it's a breaking change. Removed in my recent commit.
> * [ ]  The new Icon sizes aren't usable in React, and also aren't added to the `html-data.json`
- I don't know how to add them fully, but I created the new icon sizes to fix the alignment issues we have across icons. Can you help me make these usable in react and add them to the `html-data.json`?

BumbleB2na reviewed (commented):
Missing some test coverage, mostly where code was changed outside of css styles block. Please format all files using Prettier to clean them up.

twjeffery commented on libs/web-components/src/components/app-header-menu/AppHeaderMenu.svelte:
```diff
@@ -202,118 +197,279 @@
 
 <style>
+  :host {
+/* TODO: Component tokens, to move to design tokens file ------------------------------------------------------- */
```
Updated all the files and moved published as design tokens on [NPM](https://www.npmjs.com/package/@abgov/design-tokens/v/1.5.2?activeTab=versions)

twjeffery commented on libs/web-components/src/components/app-header/AppHeader.svelte:
```diff
@@ -278,30 +278,177 @@
 
 <style>
+  :host {
+    /* TODO: Component tokens, to move to design tokens file ------------------------------------------------------- */
```
removed

twjeffery commented on libs/web-components/src/components/button/Button.svelte:
```diff
@@ -88,59 +88,134 @@
-    box-sizing: border-box;
     cursor: pointer;
-    font: var(--goa-button-text);
+    font: var(--goa-button-default-typography);
```
Good catch. I updated these to use the same tokens used with the privacy portal team. Should all be the same now.

twjeffery commented on libs/web-components/src/components/button/Button.svelte:
```diff
@@ -151,134 +226,196 @@
       width: 100%;
       display: flex;
     }
+    button.tertiary {
```
for some reason, yes.

twjeffery commented on libs/web-components/src/components/button/Button.svelte:
```diff
@@ -151,134 +226,196 @@
-    font-size: var(--goa-font-size-4);
-    padding-left: var(--goa-button-padding-lr-compact);
-    padding-right: var(--goa-button-padding-lr-compact);
+    height: var(--goa-button-compact-height);
```
updated to align with the initial flagship tokens, all the tokens for privacy portal should be the same now.

twjeffery commented on libs/web-components/src/components/calendar/Calendar.svelte:
```diff
@@ -311,7 +311,8 @@
         data-testid="months"
-        width="calc(314px / 2 - 1.5rem)"
+        width="160px"
+        maxHeight="240px"
```
both seem to work. I've updated it to use `maxheight`

twjeffery commented on libs/web-components/src/components/calendar/Calendar.svelte:
```diff
@@ -327,7 +328,8 @@
         data-testid="years"
-        width="calc(314px / 2 - 1.5rem)"
+        width="104px"
+        maxHeight="240px"
```
updated to use `maxheight`

twjeffery commented on libs/web-components/src/components/container/Container.svelte:
```diff
@@ -181,84 +205,98 @@
 
   .padding--relaxed header {
-    padding: 0 var(--goa-container-padding);
+    padding: 0 var(--goa-container-relaxed-padding);
```
updated to align with the initial flagship tokens, all the tokens for privacy portal should be the same now.

twjeffery commented on libs/web-components/src/components/dropdown/Dropdown.svelte:
```diff
@@ -84,6 +83,9 @@
   let _error = toBoolean(error);
   let _prevError = _error;
 
+  let _filterable = toBoolean(filterable) && !toBoolean(native);
```
removed

twjeffery commented on libs/web-components/src/components/side-menu-group/SideMenuGroup.svelte:
```diff
@@ -206,19 +266,19 @@
-    margin-left: 1rem;
-    border-left: var(--goa-side-menu-child-border-width) solid var(--goa-color-greyscale-100);
-    padding: 0.5rem 1rem 0.5rem 1rem;
+    margin-left: var(--goa-side-menu-sub-item-margin);
```
updated to align with the initial flagship tokens, all the tokens for privacy portal should be the same now.

twjeffery commented on libs/web-components/src/components/side-menu/SideMenu.svelte:
```diff
@@ -106,30 +106,80 @@
   :global(::slotted(a:visited)) {
     /* required to override base styles */
-    color: var(--goa-side-menu-text-color, var(--goa-color-text-default)) !important;
+    color: var(--goa-side-menu-color-menu-item) !important;
```
updated to align with the initial flagship tokens, all the tokens for privacy portal should be the same now.

twjeffery commented on libs/web-components/src/components/app-header/AppHeader.svelte:
```diff
@@ -20,9 +20,9 @@
+    "data:image/svg+xml,%3Csvg width='118' height='32' viewBox='0 0 155 42' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cg clip-path='url(%23clip0_74_4890)'%3E%3Cpath d='M155 21H140.6V35.4004H155V21Z' fill='%2300B6ED'/%3E%3Cpath d='M65.5539 33.211C64.116 33.349 62.6147 33.4882 61.0366 33.5566C61.6106 29.2005 63.9758 23.1883 66.915 24.1484C68.6357 24.7016 67.7035 29.8197 65.5527 33.2086L65. ...
 
   let _slotParentEl: HTMLElement;
   let _rootEl: HTMLElement;
```
I used prettier to reformat all the files.

twjeffery commented on libs/web-components/src/components/app-header/AppHeader.svelte:
```diff
@@ -236,13 +236,13 @@
         <div
           slot="target"
-          style={styles("height: 100%")}
+          style={styles("height: 100%; display: flex; align-items: end;")}
```
good idea, moved to the `.tablet .menu-toggle-area {` styling

twjeffery commented on libs/web-components/src/components/badge/Badge.svelte:
```diff
@@ -96,63 +96,99 @@
   :host {
     box-sizing: border-box;
     font-family: var(--goa-font-family-sans);
+    height: 22px;
```
nope, removed.

twjeffery commented on libs/web-components/src/components/calendar/Calendar.svelte:
```diff
@@ -393,106 +395,140 @@
   .day:focus-within {
-    outline: var(--goa-border-width-l) solid var(--goa-color-interactive-focus);
+    outline: var(--goa-date-input-day-border-focus);
+    z-index: 1000;
```
I don't think this is an issue, as you can only ever focus one thing at a time? (the dropdown would never be focused and opened at the same time as the day)

BumbleB2na replied:
Oh, right. Tested and it works fine

twjeffery commented on libs/web-components/src/components/file-upload-input/FileUploadInput.svelte:
```diff
@@ -195,7 +195,7 @@
       <goa-icon type="cloud-upload" size="large" />
       <div>Drag and drop files here</div>
-      <em>or</em>
+      <em  class="or-text">or</em>
```
done

twjeffery commented on libs/web-components/src/components/icon-button/IconButton.svelte:
```diff
@@ -40,10 +40,10 @@
+    small: "var(--goa-icon-button-medium-padding)",
+    medium: "var(--goa-icon-button-medium-padding)",
+    large: "var(--goa-icon-button-large-padding)",
+    xlarge: "var(--goa-icon-button-large-padding)",
```
The padding is correct for the 4 sizes, I know it's a bit strange, but that's what we have currently.

I've moved this to styles as you described.

twjeffery commented on libs/web-components/src/components/input/Input.svelte:
```diff
@@ -273,7 +289,7 @@
 <div
   class="container"
-  style={`--width: ${width};${calculateMargin(mt, mr, mb, ml)}`}
+  style={`${containerWidth}${calculateMargin(mt, mr, mb, ml)}`}
```
Percentage is being handled differently as this allows someone to input percentage, px, or ch as a width. If they enter ch it sets the width of the input area that you can type into, if you set percentage it sets the width of the entire input (including border and padding).

BumbleB2na replied:
Okay makes sense

twjeffery commented on libs/web-components/src/components/input/Input.svelte:
```diff
@@ -139,6 +148,13 @@
+  if (_rootEl.querySelector('.leading-icon') || _rootEl.querySelector('.trailing-icon')) {
+    _rootEl.classList.add('has-icon');
+  }
+  });
```
done, added `class:has-icon={leadingicon || trailingicon}`

twjeffery commented on libs/web-components/src/components/input/Input.svelte:
```diff
@@ -633,4 +672,13 @@
+
+  /*input:-webkit-autofill {
+  box-shadow: 0 0 0px 1000px #E0F0FC inset !important;
+  } */
```
I want to add styling for the autofill. Added a `TODO` instead of removing

twjeffery commented on libs/web-components/src/components/link/Link.svelte:
```diff
@@ -35,14 +32,12 @@
 
-  :global(::slotted(a)) {
-    color: var(--goa-color-interactive-default);
-  }
 </style>
```
ok, I added it back.

twjeffery commented on libs/web-components/src/components/popover/Popover.svelte:
```diff
@@ -235,8 +237,8 @@
-    style("width", width),
-  )}
+    style("width", width)
+      )}
```
done

twjeffery commented on libs/web-components/src/components/popover/Popover.svelte:
```diff
@@ -286,39 +288,48 @@
+    font: var(--goa-typography-body-m);
     display: flex;
     align-items: center;
+    height: 100%;
```
yea

twjeffery commented on libs/web-components/src/components/popover/Popover.svelte:
```diff
@@ -286,39 +288,48 @@
   .popover-target {
     cursor: pointer;
     height: 100%;
+   /* outline: none; -> hides second blue focus, need to update tab index (-1) to skip focus */
```
done

twjeffery commented on libs/web-components/src/components/skeleton/Skeleton.svelte:
```diff
@@ -277,7 +279,6 @@
 
   .card {
-    display: inline-block;
   }
```
removed

twjeffery commented on libs/web-components/src/components/badge/Badge.svelte:
```diff
@@ -79,10 +79,10 @@
     />
   {:else}
-    <div style="height: 1.2rem; margin-left:-0.25rem;" />
+    <div style="height: 22px; margin-left:-0.25rem;" />
```
moved to a class

twjeffery commented on libs/web-components/src/components/dropdown/Dropdown.svelte:
```diff
@@ -840,25 +887,23 @@
     flex: 1 1 auto;
-    font-family: var(--goa-font-family-sans);
     z-index: 1;
+    text-overflow: ellipsis;
```
The ellipses shows only when the selected option overflows the width of the dropdown input. This is to indicate to the user that the option is being cut off. This only shows when the width of the longest option is wider than the screen/container content width. 

Max width 100% ensures that the dropdown doesn't go wider than the content/screen width in the above circumstance.

This is what happens without max-width 100% (overflow off screen width)
<img width="666" alt="image" src="https://github.com/user-attachments/assets/f4a62495-e20b-4e66-9ab7-5832b1f58eaa" />

BumbleB2na replied:
It works good! Tested it out with basic and filterable dropdowns

BumbleB2na commented on libs/web-components/src/components/calendar/Calendar.svelte:
```diff
@@ -393,106 +395,140 @@
+    border-radius: var(--goa-date-input-calendar-border-radius);
     padding: 1rem;
   }
+  */
```
Thanks. Looks like there's a default Calendar border that's turned off when used inside Datepicker. Not sure if it's used anywhere. Turning it on inside datepicker would look like this:
![image](https://github.com/user-attachments/assets/811caea1-c86a-4eb8-b426-237b8dc0534d)

twjeffery replied:
yea, we don't want that additional border.

twjeffery commented on libs/web-components/src/components/side-menu/SideMenu.svelte:
```diff
@@ -111,25 +111,29 @@
   .side-menu {
-    display: block;
-  }
+    width: var(--goa-side-menu-width);
```
ok, I don't want to introduce breaking changes for teams, I will take both `width` and `border-right` out. 

Right now we are setting both width and the right border manually on the parent div... it would be nice to move to it to the component in the future. But we can still cover this in the examples code for now too.

twjeffery replied:
I also updated the focus on the menu to inset so that it doesn't get clipped off when against the edge of the screen.

twjeffery commented on libs/web-components/src/components/side-menu-heading/SideMenuHeading.svelte:
```diff
@@ -16,16 +16,25 @@
+  goa-icon {
+    color: var(--goa-side-menu-icon-color);
+    font-size: var(--goa-side-menu-icon-size);
+    margin-right: var(--goa-side-menu-heading-icon-gap);
```
I added gap on the SideMenuHeading to ensure space when using a badge

twjeffery commented on libs/web-components/src/components/app-header/AppHeader.svelte:
```diff
@@ -311,272 +333,401 @@
+  /* Menu items in collapsed menu */
+  :global(::slotted(a.inside-collapse-menu)) {
+    color: var(--goa-app-header-color-text-nav-item);
+    background-color: var(--goa-app-header-color-bg-nav-item);
```
Ah I see that no styling was created for the current nav item when in menu. Updated and added two tokens. I think this works correctly now.

twjeffery commented on libs/web-components/src/components/side-menu-group/SideMenuGroup.svelte:
```diff
@@ -153,36 +153,48 @@
+    padding: var(--goa-side-menu-padding-child);
     text-decoration: none;
-    border-left: var(--goa-side-menu-child-border-width) solid var(--goa-color-greyscale-100);
+    border-left: var(--goa-side-menu-child-border-left);
```
Are we able to just tell them to update the tokens:
`--goa-side-menu-child-border-left: none`
`--goa-side-menu-child-border-left-selected: none`
`--goa-side-menu-child-border-left-hover: none`

in their override css?

twjeffery commented on package.json:
```diff
@@ -11,7 +11,7 @@
     "build:vscode-doc": "node libs/web-components/custom-element-manifest-analyze.js",
     "pretest": "npx nx run common:build",
-    "test": "vitest && nx test angular-components && nx test react-components",
+    "test": "vitest",
```
reverted back

twjeffery commented on libs/web-components/src/components/badge/Badge.svelte:
```diff
@@ -100,59 +100,65 @@
     display: inline-flex;
+    height: var(--goa-badge-height);
+    width: auto;
+    vertical-align: top;
```
vertically align top in the table within a row is from the updated table code (which is intentional)

chrisolsen commented on libs/web-components/src/components/dropdown/Dropdown.svelte:
```diff
@@ -643,7 +661,9 @@
       {relative}
       data-testid="option-list"
-      width={_popoverMaxWidth || "300px"}
+      width={`${_popoverMaxWidth || 0}`}
```
I will have to check on what happens when set to `0`, but that might be ok.

twjeffery commented on libs/web-components/src/components/file-upload-input/FileUploadInput.svelte:
```diff
@@ -254,56 +254,60 @@
 
 <style>
+  :host {
+    /* TODO need to add tokens for the file upload card */
```
no, I'll remove

twjeffery commented on libs/web-components/src/components/input/Input.svelte:
```diff
@@ -124,6 +126,14 @@
     });
   }
 
+  $: {
```
ok, I updated to be under onMount instead of reactive. Please confirm that I did this right

twjeffery commented on libs/web-components/src/components/tabs/Tabs.svelte:
```diff
@@ -254,70 +254,99 @@
 </div>
 
 <style>
+
```
updated to fix and use defined gap. Also vertically aligned badge with text.

twjeffery commented on libs/web-components/src/components/icon/Icon.svelte:
```diff
@@ -1,9 +1,11 @@
-  export type IconSize = "small" | "medium" | "large" | "xlarge";
-  export type IconVariant = "primary" | "secondary" | "tertiary";
-  export type IconTheme = "outline" | "filled" | "sharp";
+  import { GoabIconSize, GoabIconTheme, GoabIconVariant } from "@abgov/ui-components-common";
```
what should I do instead?

twjeffery replied:
updated in latest commit - now in the file instead of imported

twjeffery commented on package.json:
```diff
@@ -106,6 +106,7 @@
     "vitest-dom": "^0.1.1"
   },
   "dependencies": {
+    "@abgov/ui-components-common": "^1.0.0",
```
updated icon file and removed this dependancy

twjeffery commented on package-lock.json:
```diff
@@ -9,6 +9,7 @@
       "version": "0.0.0",
       "license": "Apache-2.0",
       "dependencies": {
+        "@abgov/ui-components-common": "^1.0.0",
```
not yet, just did in another commit

BumbleB2na reviewed (dismissed):
lgtm

tzuge commented:
:tada: This issue has been resolved in version 1.2.0-alpha.1 :tada:

The release is available on:
- [npm package (@alpha dist-tag)](https://www.npmjs.com/package/@abgov/ui-components-common/v/1.2.0-alpha.1)
- [GitHub release](https://github.com/GovAlta/ui-components/releases/tag/common-v1.2.0-alpha.1)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

tzuge commented:
:tada: This issue has been resolved in version 4.2.0-alpha.1 :tada:

The release is available on:
- [npm package (@alpha dist-tag)](https://www.npmjs.com/package/@abgov/angular-components/v/4.2.0-alpha.1)
- [GitHub release](https://github.com/GovAlta/ui-components/releases/tag/angular-v4.2.0-alpha.1)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

tzuge commented:
:tada: This issue has been resolved in version 1.32.0-alpha.1 :tada:

The release is available on:
- [npm package (@alpha dist-tag)](https://www.npmjs.com/package/@abgov/web-components/v/1.32.0-alpha.1)
- [GitHub release](https://github.com/GovAlta/ui-components/releases/tag/web-components-v1.32.0-alpha.1)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

tzuge commented:
:tada: This issue has been resolved in version 6.2.0-alpha.1 :tada:

The release is available on:
- [npm package (@alpha dist-tag)](https://www.npmjs.com/package/@abgov/react-components/v/6.2.0-alpha.1)
- [GitHub release](https://github.com/GovAlta/ui-components/releases/tag/react-v6.2.0-alpha.1)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

tzuge commented:
:tada: This issue has been resolved in version 1.2.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@abgov/ui-components-common/v/1.2.0)
- [GitHub release](https://github.com/GovAlta/ui-components/releases/tag/common-v1.2.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

tzuge commented:
:tada: This issue has been resolved in version 4.2.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@abgov/angular-components/v/4.2.0)
- [GitHub release](https://github.com/GovAlta/ui-components/releases/tag/angular-v4.2.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

tzuge commented:
:tada: This issue has been resolved in version 1.32.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@abgov/web-components/v/1.32.0)
- [GitHub release](https://github.com/GovAlta/ui-components/releases/tag/web-components-v1.32.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

tzuge commented:
:tada: This issue has been resolved in version 6.2.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@abgov/react-components/v/6.2.0)
- [GitHub release](https://github.com/GovAlta/ui-components/releases/tag/react-v6.2.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
